packages/reqresp/src/encoders/responseEncode.ts

```diff
@@ -28,12 +28,10 @@ export function responseEncodeSuccess(
 
       // <context-bytes> - from altair
       const contextBytes = getContextBytes(protocol.contextBytes, chunk);
-      if (contextBytes) {
-        yield contextBytes as Buffer;
-      }
+      if (contextBytes) yield Buffer.from(contextBytes);
 
       // <encoding-dependent-header> | <encoded-payload>
-      yield* writeEncodedPayload(chunk.data, protocol.encoding);
+      yield* writeEncodedPayload(Buffer.from(chunk.data), protocol.encoding);
```

Comment:
why? this PR shouldn't change `src/` unless absolutely necessary

Reply:
The `chai` assertions were passing on `Buffer` and `Uint8Array` comparison while the `vitest` is not. 

So I tried to convert all tests fixture to `Uint8Array` and then had to refactor some code to accept that type of data. 

Will go over once again to make sure nothing unnecessary change is added here.